packages/jest-mock/src/__tests__/jest_mock.test.js

```diff
@@ -180,9 +183,140 @@ describe('moduleMocker', () => {
+      expect(typeof ClassBarMock.fooProp).toBe('function');
+      expect(ClassBarMock.foo.mock).toBeDefined();
+      expect(ClassBarMock.fooProp.mock).toBeDefined();
+    });
+
+    it('mocks methods in all the prototype chain (null prototype)', () => {
+      const Foo = Object.assign(Object.create(null), {foo() {}});
+      const Bar = Object.assign(Object.create(Foo), {bar() {}});
+
+      const BarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(Bar),
+      );
+      expect(typeof BarMock.foo).toBe('function');
+      expect(typeof BarMock.bar).toBe('function');
+    });
+
+    it('does not mock methods from Object.prototype', () => {
+      const Foo = {foo() {}};
+      const Bar = Object.assign(Object.create(Foo), {bar() {}});
+
+      const BarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(Bar),
+      );
+
+      expect(BarMock).toBeInstanceOf(mockGlobals.Object);
+      expect(
+        Object.prototype.hasOwnProperty.call(BarMock, 'hasOwnProperty'),
+      ).toBe(false);
+      expect(BarMock.hasOwnProperty).toBe(
+        mockGlobals.Object.prototype.hasOwnProperty,
+      );
+    });
+
+    it('does not mock methods from Object.prototype (in mock context)', () => {
+      const Bar = vm.runInContext(
+        `
+        const Foo = { foo() {} };
+        const Bar = Object.assign(Object.create(Foo), { bar() {} });
+        Bar;
+      `,
+        mockContext,
+      );
+
+      const BarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(Bar),
+      );
+
+      expect(BarMock).toBeInstanceOf(mockGlobals.Object);
+      expect(
+        Object.prototype.hasOwnProperty.call(BarMock, 'hasOwnProperty'),
+      ).toBe(false);
+      expect(BarMock.hasOwnProperty).toBe(
+        mockGlobals.Object.prototype.hasOwnProperty,
+      );
+    });
+
+    it('does not mock methods from Function.prototype', () => {
+      class Foo {}
+      class Bar extends Foo {}
+
+      const BarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(Bar),
+      );
+
+      expect(BarMock).toBeInstanceOf(mockGlobals.Function);
+      expect(Object.prototype.hasOwnProperty.call(BarMock, 'bind')).toBe(false);
+      expect(BarMock.bind).toBe(mockGlobals.Function.prototype.bind);
+    });
+
+    it('does not mock methods from Function.prototype (in mock context)', () => {
+      const Bar = vm.runInContext(
+        `
+        class Foo {}
+        class Bar extends Foo {}
+        Bar;
+      `,
+        mockContext,
+      );
+
+      const BarMock = moduleMocker.generateFromMetadata(
+        moduleMocker.getMetadata(Bar),
+      );
+
+      expect(BarMock).toBeInstanceOf(mockGlobals.Function);
+      expect(Object.prototype.hasOwnProperty.call(BarMock, 'bind')).toBe(false);
+      expect(BarMock.bind).toBe(mockGlobals.Function.prototype.bind);
+    });
+
+    it('does not mock methods from RegExp.prototype', () => {
```

Comment:
Didn't we say we were going to remove the `RegExp` whitelisting? (i.e. either all or none).

Reply:
That would be a breaking change. `RegExp` is still a special case because we don't mock their methods (and the rest of built-ins are already managed in collections or here).